### ISSUE TITLE
chain method call should escape lua keywords

### ISSUE THREAD

**Issue by leafo** [leafo/moonscript#218]
```moonscript
x = hey.hello\if "okay"
```

expected:

```lua
local _recvr_0 = hey.hello
local x = _recvr_0["if"] _recvr_0, "okay"
```

got:

```lua
local x = hello:if("okay")
```

**Comment by leafo** [comment]
Also:

```moon
with dad
  \if "yes"
```

**Comment by s-ol** [comment]
aren't lua keywords as identifiers an issue in general?
You can write `end = 3` or `repeat = 3` or `until = 3` in moonscript and it parses without an issue, but the compiled Lua is invalid. I think it should at least throw an error or even be escaped automatically (only possible for local variables)

**Comment by leafo** [comment]
that's true, those should be syntax errors<sup>1</sup>. But we currenltly allow doing something like this:

```moonscript
hello.if = 134
```

```lua
hello["if"] = 134
```

<sup>1</sup> or implement a system that rewrites local variable names
